### ISSUE TITLE
Can't start "Back WHB" Backtesting Session on remote Ubuntu 18.04 laptop

### ISSUE THREAD

**Issue by anvd** [Superalgos/Superalgos#254]
I ran the Superalgos backend (develop branch) on a spare laptop running Ubuntu 18.04. Then I have successfully accessed the http://localhost:34247/ address from another laptop.

I started the the "Backtesting WHB" task, then when starting the "Back WHB" Backtesting session, I received the message "Session cannot be run"
![session_cannot_run](https://user-images.githubusercontent.com/12640348/83822028-2a4dd480-a6fa-11ea-93a7-50a37fb11792.jpg)

I have 6 sensors bot running pulling data from Binance exchange: 5 in Masters BTC/USDT task manager and 1 in Sparta BTC/USDT task manager. The image shows the status of bots in Masters BTC/USDT
<img width="1438" alt="sensor_bots_status" src="https://user-images.githubusercontent.com/12640348/83821944-f07cce00-a6f9-11ea-8db1-36da59557c1e.png">

Checking the Chrome console, Superalgos tries to access file at “http://localhost:34247/Storage/Binance/BTC-USDT/TradingEngines/ ….” folder on the backend side
![Chrome_Console](https://user-images.githubusercontent.com/12640348/83821780-7cdac100-a6f9-11ea-8eec-afc9f6281634.jpg)

Checking the folder structure of Superalgos source code on the spare laptop, it seems "./Storage/Binance/BTC-USDT/TradingEngines/" folder doesn't exist?
<img width="970" alt="Superalgos_folder_structure" src="https://user-images.githubusercontent.com/12640348/83822230-a6481c80-a6fa-11ea-97d2-91c57d83cb80.png">

Please help to check this. Thank you.

**Comment by Luis-Fernando-Molina** [comment]
HI, have you followed all the steps on this guide, one by one? If not please do it like in the guide, and let us know in which step something does not look like in the guide, in order to focus our attention there.  https://docs.superalgos.org/suite-set-up-each-node-on-the-network.html

**Comment by anvd** [comment]
@Luis-Fernando-Molina I think this is not really a trading farm. In this setup, I would like to run SuperAlgos on a spare laptop (like running on an AWS EC2 instance). Then on a another laptop, I use the SSH tunnel to access the frontend and control SuperAlgos. I'm a little busy at the moment. Maybe next week, I will spend time setting up everything from scratch then let you know whether the issue still exists.

**Comment by Luis-Fernando-Molina** [comment]
Someone wrote this guide and shared on the community tegram. Still not added to the documentation. It might help you:

```
#How to setup superalgos to run on a linux box (This guide is only 100% compatible with ubuntu 18.04, not sure about other versions of linux)

# 1) After getting a linux box make sure port 22 is open to allow SSH access

# 2) Open up terminal or powershell if on windows and ssh into the VPS using

	ssh username@vm_ip_address

# You may need to add -4 before username@vm_ip_address depending on how your vps is setup

# 3) Update and upgrade your repos with

	sudo apt update && sudo apt upgrade

# 4) Install tmux to allow you to exit out of the ssh connection without shutting down superalgos with. How it works is it basically opens up a shell within the linux instance which you then "stream"? I guess is the most basic way of explaining it. (Opens a shell within a shell)

	sudo apt install tmux

#5) Install nodejs

	sudo apt install nodejs
 
#6) Download the superalgos repo to your vps using git using the following 3 commands

	# 1: Install git
	sudo apt install git

	# 2: Go to root directory for easy access
	cd

	# 3: Clone the superalgos repo
	# By removing -b develop you will download the stable release
	# By keeping -b develop you will download the newest test version
	git clone -b develop https://github.com/Superalgos/Superalgos.git  


# You can now close out your SSH connection

	exit

# or you can just x out of your terminal/powershell
-----------------------------------------------------

# You should now be ready to run Superalgos using a SSH pipeline

# But before we can do that we need to setup a tmux session
# Skip these steps if you've already done this before and entering

	tmux ls

# doesnt return something like
# "no server running  blah blah"

# If it does then you need to do these steps again

# 1) You want to open a SSH connection to your VPS but you want to pipe the ports superalgos uses so use the following command

	ssh -L 34247:localhost:34247 -L 8080:localhost:8080 username@vm_ip_address

# You may once again need to add -4 before username@vm_ip_address depending on how your VPS is setup


# 2) Create a new tmux session and lets call it superalgos

	tmux new -s superalgos

# You should now be inside of a tmux session
-----------------------------------------------------

# Now that you have a tmux session running we can run superralgos!

# 1) If you arent already, SSH into your VPS

	
	ssh -L 34247:localhost:34247 -L 8080:localhost:8080 username@vm_ip_address
	
# 2) Go into your tmux session if you aren't already in it
	
	tmux -session -t name-of-session

		#Note: If you dont remember what you named it you can check by doing
			
			tmux ls

		# and it should say something along the lines of 
		# 0: 1 windows (created Sat May 30 07:24:54 2020) [202x57]
		# or
		# superalgos: 1 windows (created Sat May 30 07:24:54 2020) [202x57]

		# If it returns something along the lines of 
		# "no server running  blah blah"
		# Go to the previous section on how to make a tmux session

		# the first part of that response before the : is the name of the session

# 3) Navigate to where we downloaded superalgos
	
	cd Superalgos/

# 4) Start it up!

	node run

# You should now be able to open a chrome tab and going to 

	http://localhost:34247/

# should open up superalgos!
-----------------------------------------------------

# How do you close your terminal without shutting down superalgos? (For when you have paper/live trading or forward testing sessions that you dont want to stop)

# You need to exit your tmux session before you close your ssh connection

# do that by pressing command and b together  if youre on mac, or by pressing ctrl and b together on windows, letting go, then pressing d to detach from the tmux session while superalgos is running

# If you were to try

	tmux ls

# It should show a session that's still running

# youre now free to close out your SSH connection and close terminal/powershell and your superalgos session wouldnt shut down

# Heads up, you wont be able to use the ui until you connect back again
-----------------------------------------------------

# How do you connect back to your superalgos session thats running on your VPS in the event you need to use the ui? 

# You can do that by opening up a SSH connection again and attaching to the tmux session

# 1) SSH into your server again using terminal/powershell by typing the command 

	ssh -L 34247:localhost:34247 -L 8080:localhost:8080 username@vm_ip_address

# 2) Connect to your tmux session by doing

	tmux attach-session -t name-of-session

		# Note: If you dont remember what you named it you can check by doing
			
			tmux ls

		# and it should say something along the lines of 
		# 0: 1 windows (created Sat May 30 07:24:54 2020) [202x57]
		# or
		#superalgos: 1 windows (created Sat May 30 07:24:54 2020) [202x57]

		# If it returns something along the lines of 
		# "no server running  blah blah"
		# Go to the previous section on how to make a tmux session

		# the first part of that response before the : is the name of the session

# 3) You should see superalgos chugging along and opening a chrome tab again and going to 

	http://localhost:34247/

# Should let you interact with superalgos again!
-----------------------------------------------------

# Credits to @NerdyNigel on Telegram for the help
```

**Comment by Luis-Fernando-Molina** [comment]
Please let me know if that helped.

**Comment by anvd** [comment]
I don't have time to install everything from scratch yet. But this is exactly the guide I followed.

**Comment by julian-molina** [comment]
The first screen you shared does not show the error the session is encountering (a yellow message below the red ring). I suspect it may be related to an issue with the workspace in the latest release, which has been fixed since. The problem was that the session was not referencing the trading system. If that is the case, you may create the reference manually or update your workspace to the one in the master branch.

**Comment by anvd** [comment]
@julian-molina @Luis-Fernando-Molina I have just performed a git pull, installed everything from scratch and checked that there is a reference from the backtesting session to the trading system.

My configuration is as follows:
+ Install SuperAlgo on an Ubuntu 18.04 laptop, remove the line that opens the browser in "webServer.js" (open('http://localhost:' + port)).
+ From a Mac OS laptop, use SSH tunnel to access the front-end of SuperAlgos running on Ubuntu Laptop.

I have a question: does Superalgos support accessing the front-end from multiple machines at the same time?

I access the frontend from the Ubuntu 18.04 laptop and from the Mac OS (using SSH Tunnel). I can start the backtesting session from Chrome on the MacOS.

1. The frontend has been started from Chrome on Mac OS. But the user interface on Ubuntu Chrome does not reflect the status (the backtesting session has been started).
2. The animated green-yellow circle indicating the Backtesting session running on MacoS Chrome disappears after showing up for several seconds.

Please have a look at the following two images.

![Frontend_on_MacOS](https://user-images.githubusercontent.com/12640348/84561822-d8e8b980-ad79-11ea-8dc3-a5eda2811f77.jpg)
![Frontend_on_Ubuntu_18 04](https://user-images.githubusercontent.com/12640348/84561831-e605a880-ad79-11ea-8f4c-6de39ed0caca.jpg)

Is this normal?

**Comment by Luis-Fernando-Molina** [comment]
The system is not prepared yet to be managed from different machines at the same time. Even though it is not prepared it will allow you to do some stuff, but without working properly at the UI level.

If you run a task from laptop 1 it will show the progress, but laptop 2 will not know it is running.
If you run the same task from laptop 2 and it is already running at the backend, then the backend will ignore it, but laptop 2 will switch to 'monitor the task' and it will also display its status.

From any of the 2 laptops you can stop it and it should work.

That is how it works today.

**Comment by anvd** [comment]
@Luis-Fernando-Molina Thank you for the explanation.
